network.c

```diff
@@ -96,7 +96,8 @@ static int
 open_client_connection(char *port_name)
 {
 	char hostname[BSIZE];
-	strncpy(hostname, port_name, sizeof(hostname));
+	strncpy(hostname, port_name, sizeof(hostname)-1);
+	hostname[sizeof(hostname-1)] = 0;
```

Comment:
Maybe you meant `hostname[sizeof(hostname)-1]` ?

Reply:
Good catch! Thanks. I added more stuff to the patch. Passes `-Wall -Wextra -Wno-unused-parameter` on Linux too.